app/models/user.rb

```diff
@@ -0,0 +1,26 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: :author_id
+  has_many :comments
+  has_many :likes
+  has_many :liked_posts, through: :likes, source: :post
+  has_many :liked_comments, through: :likes, source: :comment
+  validates :name, length: { maximum: 100 }, presence: true
+  validates :bio, length: { maximum: 300 }, allow_blank: true
+  has_one :picture
+
+  before_save :default_values
+
+  def update_counters
+    update(posts_count: posts.count, likes_count: likes.count, comments_count: comments.count)
+  end
```

Comment:
- Kindly do the same on the Comment 

```ruby
def update_comments_counter
  post.increment!(:comments_counter)
end
```
and like:

```ruby
def update_likes_counter
  post.increment!(:likes_counter)
end
```

Reply:
Not required, so I will keep the method I am using now, as it keeps the count values consistent with the actual amount of instances that exist.

Reply:
Well resolved ✔️ 💯

Reply:
[OPTIONAL] It would be best if you adopted a `growth mindset` and showed a willingness to adopt new solutions if they are more effective and improve the performance of your application. 😊